test/python/pulse/test_schedule.py

```diff
@@ -431,10 +431,16 @@ def test_delay_snapshot_channel(self):
 
 
 class TestScheduleFilter(BaseTestSchedule):
-    """Test Schedule filtering methods."""
-
+    """
+    Test Schedule filtering methods: Schedule.filter and Schedule.exclude
+
+    Note: each test in this class uses the helper: self.filter_and_test_consistency
```

Comment:
It is not necessary to document this at the class level. This is an internal helper function, not really necessary for someone to understand this outright. We could even make `filter_and_test_consistency` a private method

Furthermore, comments are hard to maintain; nothing will ever break if a comment is outdated, but it may mislead developers. This documentation should therefore remain close to the function it describes. This improves the chances that we won't end up with this comment remaining in the future if things change, for instance if that helper function is removed.